Cargo.toml

```diff
@@ -42,6 +42,7 @@ wat = { workspace = true }
 serde = "1.0.94"
 serde_json = "1.0.26"
 wasmparser = { workspace = true }
+wasm-coredump-builder = { version = "0.1.9" }
```

Comment:
This is also visible on CI, but we're stricter-than-average about the dependencies we pull in for Wasmtime. Relying on something external here is fine but this specifically pulls in a number of things which I think we'll want to change/address:

* `colored` - in theory this shouldn't be necessary for just building core dumps
* `core-wasm-ast` - this looks like it may be duplicating the functionality of the already-in-use `wasmparser` crate perhaps? If that's the case we'd prefer to only have one definition of parsing wasm within wasmtime. 
* `hermit-abi` - I think this is due to the `num_cpus` update, but is the update required?
* `threadpool` - I think in theory this shouldn't be necessary for a core dump builder
* `wasm-printer` - does this perhaps duplicate the functionality already present in `wasmprinter`? If so like `wasmparser` it would be best to only have one implementation of this rather than multiple.

I'll also note that we're also already using the `wasm-encoder` crate for creating core wasm modules, so if core dump functionality could be added to that or somehow leverage that I think that would be ideal.

Reply:
`colored`, `threadpool` and `num_cpus`  are dependencies not directly used by `wasm-coredump-builder`, I'll make sure to remove them.

`core-wasm-ast` (in theory, just defines the stucts to represent coredump) and `wasm-printer` (maybe a bad name, it's used to serialize the coredump to binary Wasm) are required by `wasm-coredump-builder`. 

> I'll also note that we're also already using the wasm-encoder crate for creating core wasm modules, so if core dump functionality could be added to that or somehow leverage that I think that would be ideal.

That's indeed duplicated features with my crates. I believe I can rewrite `wasm-coredump-builder` to only use `wasm-encoder`. A disadvantage of that approach on my side is that it won't integrate well with my tools / crates, but could be specific to wasmtime.

Reply:
With the libararified / modular CLI approach, dependecies wouldn't matter as much right?

Reply:
To clarify I don't mean to pass judgement on which one should be used per se, but within the context of Wasmtime's dependency graph we'd prefer to keep only one implementation of things rather than bringing together various islands of ecosystems. I don't mean to say that you should rewrite crates on top of the crates in the `wasm-tools` repository (although feedback on those is always appreciated), but I believe that `wasm-encoder` is high-level enough and the core dump format is simple enough that doing it "inline" as opposed to using an external crate wouldn't be the end of the world (not that using an external crate is wrong, again just trying to keep the dependencies in Wasmtime under control)

Reply:
yeah I understand, I wasn't trying to argue against. I'm planning to rewrite `wasm-coredump-builder` to depend only on a new coredump specific encoding crate + `wasm-encoder`. That should address the concern.

Reply:
I published a new version of `wasm-coredump-builder` that depends on `wasm-encoder` and two coredump specific crates:
- https://github.com/xtuc/wasm-coredump/tree/main/lib/coredump-encoder
- https://github.com/xtuc/wasm-coredump/tree/main/lib/coredump-types

I'm not sure why `num_cpus` has been updated. Is it still related to my change?